lib/tasks/bundlers/generateLibraryPreload.js

```diff
@@ -326,32 +293,32 @@ module.exports = function({workspace, taskUtil, options: {projectName, excludes
 					filters = ["jquery.sap.global.js"];
 				}
 				p = Promise.all([
-					moduleBundler({
+					execModuleBundlerIfNeeded({
 						options: getModuleBundlerOptions({name: "sap-ui-core.js", filters, preload: true}),
 						resources
 					}),
-					moduleBundler({
+					execModuleBundlerIfNeeded({
 						options: getModuleBundlerOptions({
 							name: "sap-ui-core-dbg.js", filters, preload: false,
 							moduleNameMapping: unoptimizedModuleNameMapping
 						}),
 						resources: unoptimizedResources
 					}),
-					moduleBundler({
+					execModuleBundlerIfNeeded({
 						options: getModuleBundlerOptions({
 							name: "sap-ui-core-nojQuery.js", filters, preload: true, provided: true
 						}),
 						resources
 					}),
-					moduleBundler({
+					execModuleBundlerIfNeeded({
 						options: getModuleBundlerOptions({
 							name: "sap-ui-core-nojQuery-dbg.js", filters, preload: false, provided: true,
 							moduleNameMapping: unoptimizedModuleNameMapping
 						}),
 						resources: unoptimizedResources
 					}),
 				]).then((results) => {
-					const bundles = Array.prototype.concat.apply([], results);
+					const bundles = Array.prototype.concat.apply([], results).filter(Boolean);
```

Comment:
Is the concat still needed? Promise.all always resolves with an array and the filter(Boolean) anyhow creates a new array.

Reply:
Or is this meant to flatten the results ? `Array.prototype.flat` should be available since NodeJS 11 ...

Reply:
Yes, it should flatten the results. We do this in several places, so maybe we should do some general code-cleanup change.

Reply:
Yes, this is a common pattern in the UI5 Tooling code base. We should clean those things up in one sweep

Reply:
okay with me.